lib/usd/translators/shading/usdOpenPBRSurfaceWriter.cpp

```diff
@@ -0,0 +1,420 @@
+#include <pxr/pxr.h>
+#include <pxr/usd/sdf/valueTypeName.h>
+#include <pxr/usd/usdShade/shader.h>
+#include <pxr/usd/usdShade/tokens.h>
+
+#include <maya/MFnDependencyNode.h>
+#include <maya/MPlug.h>
+#include <maya/MStatus.h>
+
+#include <basePxrUsdPreviewSurface/usdPreviewSurface.h>
+
+#include <cmath>
+#include <type_traits>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+namespace {
+
+template <class T> T _GetMayaAttributeValue(const MPlug& attrPlug)
+{
+    SdfValueTypeName typeName = SdfValueTypeNames->Float;
+    if (std::is_same<T, GfVec3f>()) {
+        typeName = SdfValueTypeNames->Color3f;
+    }
+
+    VtValue value = UsdMayaWriteUtil::GetVtValue(attrPlug, typeName);
+
+    if (!TF_VERIFY(
+            !value.IsEmpty() && value.IsHolding<T>(),
+            "No value found for '%s'. Incorrect type?\n",
+            MFnAttribute(attrPlug.attribute()).name().asChar())) {
+        return {};
+    }
+
+    return value.UncheckedGet<T>();
+}
+
+template <class T>
+T _GetMayaAttributeValue(const MFnDependencyNode& depNodeFn, const TfToken& attrName)
+{
+    MStatus status;
+    MPlug   attrPlug = depNodeFn.findPlug(
+        depNodeFn.attribute(attrName.GetText()),
+        /* wantNetworkedPlug = */ true,
+        &status);
+    if (!TF_VERIFY(
+            status == MS::kSuccess && !attrPlug.isNull(),
+            "Invalid plug for attribute '%s'\n",
+            MFnAttribute(attrPlug.attribute()).name().asChar())) {
+        return {};
+    }
+
+    return _GetMayaAttributeValue<T>(attrPlug);
+}
+
+void _AuthorEmissionAndDiffuse(
+    const MFnDependencyNode& depNodeFn,
+    UsdShadeShader&          shaderSchema,
+    const UsdTimeCode        usdTime)
+{
+    // All these OpenPBR attribute contribute to Diffuse:
+    const auto baseWeight
+        = _GetMayaAttributeValue<float>(depNodeFn, TrMayaOpenPBRTokens->baseWeight);
+    const auto baseColor
+        = _GetMayaAttributeValue<GfVec3f>(depNodeFn, TrMayaOpenPBRTokens->baseColor);
+    const auto baseMetalness
+        = _GetMayaAttributeValue<float>(depNodeFn, TrMayaOpenPBRTokens->baseMetalness);
+    const auto specularWeight
+        = _GetMayaAttributeValue<float>(depNodeFn, TrMayaOpenPBRTokens->specularWeight);
+    const auto subsurfaceWeight
+        = _GetMayaAttributeValue<float>(depNodeFn, TrMayaOpenPBRTokens->subsurfaceWeight);
+    const auto subsurfaceColor
+        = _GetMayaAttributeValue<GfVec3f>(depNodeFn, TrMayaOpenPBRTokens->subsurfaceColor);
+    const auto coatWeight
+        = _GetMayaAttributeValue<float>(depNodeFn, TrMayaOpenPBRTokens->coatWeight);
+    const auto coatColor
+        = _GetMayaAttributeValue<GfVec3f>(depNodeFn, TrMayaOpenPBRTokens->coatColor);
+    const auto coatIOR = _GetMayaAttributeValue<float>(depNodeFn, TrMayaOpenPBRTokens->coatIOR);
+    const auto coatDarkening
+        = _GetMayaAttributeValue<float>(depNodeFn, TrMayaOpenPBRTokens->coatDarkening);
+
+    // Diffuse: Converting from OpenPBR to StandardSurface requires a bit of math to compute coat
```

Comment:
Very differenct from StandardSurface here. Code was found in the MaterialX converter networks that remap OpenPBR surface to Standard surface, chained into the Standard surface to UsdPreviewSurface network.

There are a few tweaks here that should be backported to usdStandardSurfaceWriter.cpp:
 - How coat affects emissionColor
 - How subsurface affects diffuseColor